test/tftp/tftp_server_test.rb

```diff
@@ -0,0 +1,110 @@
+require 'test_helper'
+require 'tftp/server'
+require 'tftp/tftp_plugin'
+
+module TftpGenericServerSuite
+  def setup
+    @rootdir = "/some/root"
+    @mac = "aa:bb:cc:dd:ee:ff"
+    @content = "file content"
+    FileUtils.stubs(:mkdir_p).returns(true)
+    Proxy::TFTP::Plugin.settings.stubs(:tftproot).returns(@rootdir)
+    setup_paths
+  end
+
+  def pxe_config_files
+    @pxe_config_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def pxe_default_files
+    @pxe_default_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def test_set
+    pxe_config_files.each do |file|
+      @subject.stubs(:write_file).with(file, @content)
+    end
+    @subject.expects(:write_file).with(pxe_config_files.first, @content).once
```

Comment:
This doesn't look right to me -- first you stub two calls, and then you set an expectation on one of them. I *think* what you meant here was to set expectations on two "write_file" calls that must be called...